platform/troubleshoot/tsnet-connectivity.mdx

```diff
@@ -0,0 +1,190 @@
+
+- **Firewall restrictions**: Corporate firewall rules or cloud security groups might block the required ports and protocols for TSNet communication.
+
+
+## Solution
+
+To resolve TSNet connection failures, disable direct connection mode and configure TSNet to use the platform agent for communication. This approach works better in restricted network environments.
+
+<Flow id="Resolve TSNet connection failures">
+
+<Step title="Configure TSNet to disable direct connections">
+
+Configure TSNet to disable direct connections
+
+Update your vCluster configuration to explicitly disable direct peer-to-peer connections:
+
+```yaml title="vcluster.yaml"
+controlPlane:
+  statefulSet:
+    env:
+      - name: "TS_DEBUG_DIAL_DIRECT"
+        value: "false"
+```
+</Step>
+
+<Step title="Apply the configuration">
+
+Update your existing vCluster with the new configuration:
+
+```bash
+vcluster create <vcluster-name> -f vcluster.yaml --upgrade
+```
+
+Or if using Helm directly:
+
+```bash
+helm upgrade <vcluster-name> vcluster --repo https://charts.loft.sh -f vcluster.yaml -n <namespace>
+```
+</Step>
+
+<Step title="Restart the vCluster pods">
+
+Force a restart of the vCluster pods to ensure the new environment variable takes effect:
+
+```bash
+kubectl -n <namespace> rollout restart statefulset <vcluster-name>
+```
+</Step>
+
+<Step title="Verify network connectivity">
+
+Ensure the vCluster can reach the platform coordination server:
+
+```bash
+kubectl -n <namespace> exec -it <vcluster-pod> -- nslookup <your-platform-host>
+```
+</Step>
+</Flow>
+
+## Verification
+
+After completing the solution steps:
+
+<Flow>
+<Step title="Check vCluster pod status">
+
+  Ensure the vCluster pod is running in your namespace:
+
+  ```bash
+  kubectl -n <namespace> get pods | grep <vcluster-name>
+  ```
+</Step> 
+
+<Step title="Verify TSNet connectivity"> 
+
+  Check the pod logs for TSNet connection status:
+
+  ```bash
+  kubectl -n <namespace> logs <vcluster-pod> | grep -i "tsnet" | tail -10
+  ```
+</Step> 
+
+<Step title="Confirm platform status"> 
+
+  Open the vCluster Platform UI and confirm that the vCluster is marked as **Ready**. 
+
+</Step> 
+
+<Step title="Test vCluster"> 
+
+  Connect to the virtual cluster and check node access:
+
+  ```bash
+  vcluster connect <vcluster-name> -n <namespace>
+  kubectl get nodes
+  ```
+</Step>
+</Flow> 
+
+## Prevention
+
+To prevent future TSNet connection issues, you can apply the following preventive measures in your vCluster configuration:
+
+```yaml title="vcluster.yaml"
+controlPlane:
+  statefulSet:
+    env:
+      - name: "TS_DEBUG_DIAL_DIRECT"
+        value: "false"
+      - name: "TS_DEBUG_NETCHECK"
+        value: "true"
+    resources:
+      requests:
+        memory: "256Mi"
+        cpu: "100m"
+      limits:
+        memory: "512Mi"
+        cpu: "500m"
```

Comment:
Optional: You could break this resource stuff into separate step, or into best-practices section, because it seems largely unrelated, just a generic safeguard measure.